flatnav/util/VisitedNodesHandler.h

```diff
@@ -0,0 +1,175 @@
+#pragma once
+
+#include <flatnav/util/SIMDDistanceSpecializations.h>
+
+#include <cstring>
+#include <iostream>
+#include <memory>
+#include <mutex>
+#include <stdint.h>
+#include <vector>
+
+namespace flatnav {
+
+class VisitedNodesHandler {
+private:
+  uint32_t _mark;
+  uint32_t *_table;
+  uint32_t _table_size;
+
+public:
+  VisitedNodesHandler(const uint32_t size) : _mark(0), _table_size(size) {
+    // initialize values to 0
+    _table = new uint32_t[_table_size]();
+  }
+
+  inline void prefetch(const uint32_t num) const {
+#ifdef USE_SSE
+    _mm_prefetch((char *)_table[num], _MM_HINT_T0);
+#endif
+  }
+
+  inline uint32_t getMark() const { return _mark; }
+
+  inline void insert(const uint32_t num) { set(num); }
+
+  inline void set(const uint32_t num) { _table[num] = _mark; }
+
+  inline uint32_t size() const { return _table_size; }
+
+  inline void reset(const uint32_t num) { _table[num] = _mark + 1; }
+
+  inline void clear() { _mark++; }
```

Comment:
What is the difference between reset and clear? Could use a doc-string.

Reply:
some of these methods are actually dead-code. `reset` is one of them. I'll clean this up.